samcli/commands/build/utils.py

```diff
@@ -0,0 +1,106 @@
+"""
+Utilities for sam build command
+"""
+import pathlib
+from enum import Enum
+from typing import List
+
+import click
+
+from samcli.lib.build.workflow_config import CONFIG, get_workflow_config
+from samcli.lib.providers.provider import ResourcesToBuildCollector
+
+
+class MountMode(Enum):
+    """
+    Enums that represent mount mode used when build lambda functions/layers inside container
+    """
+
+    READ = "READ"
+    WRITE = "WRITE"
+
+    @classmethod
+    def values(cls) -> List[str]:
+        """
+        A getter to retrieve the accepted value list for mount mode
+
+        Returns: List[str]
+            The accepted mount mode list
+        """
+        return [e.value for e in cls]
+
+
+def prompt_user_to_enable_mount_with_write_if_needed(
+    resources_to_build: ResourcesToBuildCollector,
+    base_dir: str,
+) -> bool:
+    """
+    First check if mounting with write permissions is needed for building inside container or not. If it is needed, then
+    prompt user to choose if enables mounting with write permissions or not.
+
+    Parameters
+    ----------
+    resources_to_build:
+        Resource to build inside container
+
+    base_dir : str
+        Path to the base directory
+
+    Returns
+    -------
+    bool
+        True, if user enabled mounting with write permissions.
+    """
+
+    for function in resources_to_build.functions:
+        code_uri = function.codeuri
+        if code_uri:
+            runtime = function.runtime
+            code_dir = str(pathlib.Path(base_dir, code_uri).resolve())
+            # get specified_workflow if metadata exists
+            metadata = function.metadata
+            specified_workflow = metadata.get("BuildMethod", None) if metadata else None
+            config = get_workflow_config(runtime, code_dir, base_dir, specified_workflow)
+            # at least one function needs mount with write, return with prompting
+            if config.must_mount_with_write_in_container:
+                return prompt(config, code_dir)
```

Comment:
If you use early-exit in if statements, we can remove some indentation here to remove some complexity while reading this code (see: https://medium.com/swlh/return-early-pattern-3d18a41bba8)
```suggestion
        if not code_uri:
            continue
        runtime = function.runtime
        code_dir = str(pathlib.Path(base_dir, code_uri).resolve())
        # get specified_workflow if metadata exists
        metadata = function.metadata
        specified_workflow = metadata.get("BuildMethod", None) if metadata else None
        config = get_workflow_config(runtime, code_dir, base_dir, specified_workflow)
        # at least one function needs mount with write, return with prompting
        if not config.must_mount_with_write_in_container:
            continue
        return prompt(config, code_dir)
```

Same can be done with the other loop for layer below